Palace/MyBooks/MyBooksDownloadCenter.swift

```diff
@@ -451,69 +452,32 @@ extension MyBooksDownloadCenter {
-        NSLog("Unsupported content type for deletion.")
+        Log.warn(#file, "Unsupported content type for deletion.")
       }
     } catch {
-      NSLog("Failed to remove local content for book with identifier \(identifier): \(error.localizedDescription)")
+      Log.error(#file, "Failed to remove local content for book with identifier \(identifier): \(error.localizedDescription)")
     }
   }
 
-  func deleteLocalContent(forAudiobook book: TPPBook, at bookURL: URL) {
-#if FEATURE_OVERDRIVE
-    if book.distributor == OverdriveDistributorKey,
-        let data = try? Data(contentsOf: bookURL),
-        let json = try? JSONSerialization.jsonObject(with: data, options: []),
-        let jsonDict = json as? [String: Any] {
-      handleOverdriveContentDeletion(jsonDict, for: book)
-    }
-#endif
-    
+  private func deleteLocalAudiobookContent(forAudiobook book: TPPBook, at bookURL: URL) throws {
 #if LCP
-    if LCPAudiobooks.canOpenBook(book) {
-      deleteLCPLocalContent(for: book, at: bookURL)
-    } else {
-      deleteNonLCPLocalContent(at: bookURL)
-    }
+    let isLcpAudiobook = LCPAudiobooks.canOpenBook(book)
 #else
-    deleteNonLCPLocalContent(at: bookURL)
+    let isLcpAudiobook = false
 #endif
-  }
-  
-  private func handleOverdriveContentDeletion(_ dict: [String: Any], for book: TPPBook) {
-    // Implement specific deletion logic for Overdrive content if required.
-  }
-  
-#if LCP
-  private func deleteLCPLocalContent(for book: TPPBook, at bookURL: URL) {
-    LCPAudiobooks(for: bookURL)?.contentDictionary { dict, error in
-      guard error == nil, let dict = dict as? [String: Any] else { return }
-      self.deleteAudiobookContent(dict, for: book, bookURL: bookURL)
-    }
-  }
-  
-  private func deleteAudiobookContent(_ dict: [String: Any], for book: TPPBook, bookURL: URL) {
-    // Assuming we create and use an audiobook object to manage deletion
-    guard let jsonData = try? JSONSerialization.data(withJSONObject: dict, options: []),
-          let manifest = try? Manifest.customDecoder().decode(Manifest.self, from: jsonData),
-          let audiobook = AudiobookFactory.audiobook(
-            for: manifest,
-            bookIdentifier: book.identifier,
-            decryptor: nil,
-            token: book.bearerToken
-          ) else {
-      return
-    }
-    audiobook.deleteLocalContent(completion: { _, _ in })
-    if FileManager.default.fileExists(atPath: bookURL.path) {
-      try? FileManager.default.removeItem(at: bookURL)
+
+    // LCP Audiobooks are a single binary file, without an easily loaded manifest. So they skip this logic
+    // that deleted the local audio files, used by other audiobook types.
+    if (!isLcpAudiobook) {
+      let manifestData = try Data(contentsOf: bookURL)
+      let manifest = try Manifest.customDecoder().decode(Manifest.self, from: manifestData)
+      AudiobookFactory.audiobookClass(for: manifest).deleteLocalContent(manifest: manifest, bookIdentifier: book.identifier)
```

Comment:
This is very similar to `deleteAudiobookContent` function that we previously had, however once the `nil` account issue was fixed, it exposed that creating an `Audiobook` object was creating a player, this caused the audiobook to start playing, which then caused `deleteLocalContent` to fail. So I moved `deleteLocalContent` out to a class method, so it can be called without creating an `Audiobook` first.